contracts/extensions/MultisigPermissions.sol

```diff
@@ -81,10 +86,14 @@ contract MultisigPermissions is
 
   uint256 motionCount;
   mapping(uint256 => Motion) motions;
-  // Motion Id => User => Approvals
-  mapping(uint256 => mapping(address => bytes32)) motionApprovals;
-  // Motion Id => Permission => Approval Count
-  mapping(uint256 => mapping(ColonyRole => uint256)) motionRoleApprovalCount;
+
+  uint256 constant APPROVAL = 1;
+  uint256 constant REJECTION = 2;
```

Comment:
Broadly, I think this should be implemented using an enum:

```
enum Vote { None, Approval, Rejection }
```

And then the rest of the functionality should orient around `Vote` values.

Specifically, I think the approach of having separate `changeApproval` and `changeRejection` functions is unecessary. A single `changeVote` function should suffice, and would allow you to avoid passing in two variables (`_type` and `_setTo`), which I interpret as an artifact of trying to attach rejection functionality on top of an existing implementation. If we're going to support approvals and rejections then we should write the extension as though that had been a goal from the start.

I expect you could achieve the same goals of being able to handle vote "changes" by checking new `Vote` values against existing `Vote` values, if any, and making `_setTo` an internal value (along the lines of `voteChanged`) and using that down-stream.

Reply:
I've added the struct, and adjusted the interface to be more in line with what you suggest here, though the underlying functionality remains much the same (which, if I understand your suggestion here, I don't think there's much I can do about).

The main difference in functionality between my original implementation and this one is that you cannot simultaneously reject and approve a motion, which does make sense (though is gassier).